packages/asc-ui/src/theme/CompactThemeProvider.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { ThemeProvider, Theme } from '..'
+
+// These are the overrides for showing compact text for several typography types
+const overridesCompactTheme: Partial<Theme.ThemeInterface> = {
+  typography: {
+    h2: {
+      fontSize: '24px',
+      lineHeight: '30px',
+    },
+    h3: {
+      fontSize: '18px',
+      lineHeight: '23px',
+    },
+    h4: {
+      fontSize: '16px',
+      lineHeight: '20px',
+    },
+    p: {
+      fontSize: '16px',
+      lineHeight: '20px',
+    },
+  },
+}
+
+/**
+ * This can be used to decorate a component in order to use this specific theme just for it
+ *
+ *  @example <caption>Decorate a component to enfoce this theme</caption>
+ *  <CompactThemeProvider>
```

Comment:
Plz write something like this, because a self closed DecoratedComponent doesn't make sense.
```jsx
<CompactThemeProvider>
  <Heading>I am a header with a default style</Heading>
  <DecoratedComponent>
    <Heading>I have a compact style</Heading>
  </DecoratedComponent>
</CompactThemeProvider>
```

Reply:
This is indeed better. Thanks!